ql/src/test/results/clientpositive/llap/dynamic_partition_pruning.q.out

```diff
@@ -4277,21 +4277,37 @@ STAGE PLANS:
                   alias: srcpart
                   filterExpr: ds is not null (type: boolean)
                   Statistics: Num rows: 2000 Data size: 389248 Basic stats: COMPLETE Column stats: COMPLETE
-                  Group By Operator
-                    keys: ds (type: string)
-                    minReductionHashAggr: 0.99
-                    mode: hash
-                    outputColumnNames: _col0
-                    Statistics: Num rows: 2 Data size: 368 Basic stats: COMPLETE Column stats: COMPLETE
-                    Reduce Output Operator
-                      key expressions: _col0 (type: string)
-                      null sort order: z
-                      sort order: +
-                      Map-reduce partition columns: _col0 (type: string)
+                  Filter Operator
+                    predicate: ds is not null (type: boolean)
```

Comment:
I see 2 filter operators doing the same in this plan - which will be merged by the "downstream merge" patch.

However; to my best knowledge the TS filterExpr should only be considered "best-effort" because the reader may decide to not filter by some parts of the expr.

In this plan we should probably have removed 2 extra srcpart scans by the `SubTree` logic - that should be investigated - I've opened HIVE-24268